### DISCUSSION TITLE
Improve test matrix to include all supported spring versions

### DISCUSSION THREAD

**Discussion by stliu** [Microsoft/spring-cloud-azure#930] [Ideas]
Currently Spring Cloud Azure support multi Spring Boot and Spring Cloud versions, but those versions are not actively tested.

we need to figure out how to enable a matrix tests to actively test all support spring versions and Spring Cloud Azure libraries.

**Comment by saragluna** [comment]
### Facts:
- Spring projects release monthly.
- Several types of libraries can be released from Spring projects, snapshot, prerelease, and GA.
- We use multiple Spring projects in Spring Cloud Azure: Spring Framework, Spring Boot, Spring Data, Spring Integration, Spring Cloud Stream, etc.

### Expectations:
**The Spring Cloud Azure dev branch should align with Spring Boot on the current Spring Boot version.**
- Spring Cloud Azure's source code dependency version should rely on the return value from `https://spring.io/project_metadata/spring-boot` with `current: true`. 
- When the current version from `https://spring.io/project_metadata/spring-boot` changes, we should catch up ASAP.

**The Spring Cloud Azure dev branch should maintain a test matrix for dependent Spring projects.** -- UT
- Spring Cloud Azure defines their supported Spring Boot versions, 
  - The supported versions change along with Spring releases.
  - Such as `supported-versions.json`, listing 2.5.1 and 2.6.1 for example. 
- Spring Cloud Azure tests on each milestone and GA releases of dependent Spring projects.

**The Spring Cloud Azure should have integration tests, containing all Spring Cloud Azure libraries, to run against each version.** -- IT
- Ideally, we can call Azure Spring Initializr to generate such a sample project and run it, expecting it throws no exceptions.
- If not the ideal case, we can write such a sample project (probably we put it in the SDK repo) and run it against each version. 

### Ideas
- The Spring Cloud versions, should be parsed from API like `curl -G https://start.spring.io/pom.xml -d packaging=war -o pom.xml` and retrieved automatically
- The version matrix should be persistent somewhere, such as spring-cloud-azure-tools repo.
